### PR TITLE
Providing access to the player's LoginResult on LoginEvent

### PR DESCRIPTION
**Pull request by phenomax**
Adds the feature requested in #154

### CODE REVIEW

File: BungeeCord-Patches/0044-Providing-access-to-the-player-s-LoginResult-on-Logi.patch [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+rename to api/src/main/java/net/md_5/bungee/connection/LoginResult.java
+diff --git a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
+index f9cfeb67..24f73820 100644
+--- a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
++++ b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
+@@ -3,16 +3,6 @@ package net.md_5.bungee.connection;
+ import com.google.common.base.Charsets;
+ import com.google.common.base.Preconditions;
+ import com.google.gson.Gson;
+-import java.math.BigInteger;
```
Comment (jamierocks) [inline BungeeCord-Patches/0044-Providing-access-to-the-player-s-LoginResult-on-Logi.patch]: Please resolve these moved imports.

File: BungeeCord-Patches/0044-Providing-access-to-the-player-s-LoginResult-on-Logi.patch [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ import net.md_5.bungee.api.plugin.Cancellable;
++import net.md_5.bungee.connection.LoginResult;
+ 
+ /**
+  * Event called to represent a player logging in.
+@@ -34,9 +35,18 @@ public class LoginEvent extends AsyncEvent<LoginEvent> implements Cancellable
+      */
+     private final PendingConnection connection;
+ 
+-    public LoginEvent(PendingConnection connection, Callback<LoginEvent> done)
```
Comment (jamierocks) [inline BungeeCord-Patches/0044-Providing-access-to-the-player-s-LoginResult-on-Logi.patch]: There is going to need to be a constructor with the original signature.

**Comment by Janmm14** [comment]
I suggest to remove unneeded import order changing and possibly to not change imports at all but use fully qualified class names for mich easier patch compatibility to upstream changes.

**Comment by phenomax** [comment]
Do you think deleting unused will have so much impact? I didnt do that intentionally. It's just part of IntelliJ's clean up

**Comment by jamierocks** [comment]
Don't remove unused, only add imports.
It ensures that the patches can be merged easily.

**Review by jamierocks — CHANGES_REQUESTED** [review]
forgot that these need to be submitted -_-

**Comment by phenomax** [comment]
Thanks for reviewing! I'll apply the changes asap

File: BungeeCord-Patches/0044-Providing-access-to-the-player-s-LoginResult-on-Logi.patch (hunk 1 above)
Comment (phenomax) [reply BungeeCord-Patches/0044-Providing-access-to-the-player-s-LoginResult-on-Logi.patch]: Do you have any tip to disable IntelliJ's auto import moving?

Comment (jamierocks) [reply BungeeCord-Patches/0044-Providing-access-to-the-player-s-LoginResult-on-Logi.patch]: ye `Preferences`->Search for Imports->Disable the option

**Comment by jamierocks** [comment]
Okay then.

**Comment by phenomax** [comment]
I hate it,, when there are tons of revers in the commit history. So I ll start this PR from scratch in the next days.
@jamierocks Dont worry ^^

**Comment by jamierocks** [comment]
Thats no issue, the commits can be squashed - but whatever I guess.

**Pull request closed without merging**